exonum/src/runtime/mod.rs

```diff
@@ -251,17 +246,22 @@ pub trait Runtime: Send + Debug + 'static {
 
     /// Calls `before_commit` for all the services stored in the runtime.
     ///
-    /// # Notes for Runtime Developers.
+    /// # Guarantees
     ///
-    /// * The order of services during invocation of this method must be the same for each node.
-    /// In other words, the order of the runtime services must be the same for each node.
+    /// - Calls are ordered by increasing `instance_id`.
+    /// - Each `before_commit` call is isolated with a separate checkpoint. A call that returns
+    ///   an error will be rolled back.
     ///
     /// # Policy on Panics
     ///
     /// * Catch each kind of panics except for `FatalError` and write
```

Comment:
If the core becomes responsible for roll backs, I don't think this method shall catch panics.